src/databases/VTK/avtGEOSFileReader.C

```diff
@@ -0,0 +1,805 @@
+//  Method: avtGEOSFileReader constructor
+//
+//  Programmer: biagas2 -- generated by xml2avt
+//  Creation:   Wed Dec 4 14:55:21 PST 2024
+//
+// ****************************************************************************
+
+avtGEOSFileReader::avtGEOSFileReader(const char *filename, const DBOptionsAttributes *opt)
+    : avtVTKFileReader(filename, opt),
+      timeSteps(),
+      timeStepFileNames(),
+      rootFile(filename)
+{
+    currentTS = -1;    
+    haveReadRoot = false;
+    size_t pos = rootFile.find_last_of('.');
+    string ext = rootFile.substr(pos+1);
+    rootIsPVD = (ext == "pvd");
+    if(!rootIsPVD)
+    {
+       baseDir = FileFunctions::Dirname(rootFile) + VISIT_SLASH_STRING;
+       string baseName = FileFunctions::Basename(rootFile);
+       timeStepFileNames.push_back(baseName);
+    }
+}
+
+// ****************************************************************************
+//  Method: avtGEOSFileReader destructor
+//
+//  Programmer: biagas2 -- generated by xml2avt
+//  Creation:   Wed Dec 4 14:55:21 PST 2024
+//
+// ****************************************************************************
+
+avtGEOSFileReader::~avtGEOSFileReader()
+{
+    FreeUpResources();
+}
+
+// ****************************************************************************
+//  Method: avtGEOSFileReader::GetNTimesteps
+//
+//  Purpose:
+//      Tells the rest of the code how many timesteps there are in this file.
+//
+//  Programmer: biagas2 -- generated by xml2avt
+//  Creation:   Wed Dec 4 14:55:21 PST 2024
+//
+// ****************************************************************************
+
+int
+avtGEOSFileReader::GetNTimesteps(void)
+{
+    if(!haveReadRoot)
+        ReadRootFile();
+
+    return int(timeSteps.size());
+}
+
+// ****************************************************************************
+//  Method: avtGEOSFileReader::GetTimes
+//
+//  Purpose:
+//      Tells the rest of the code how many timesteps there are in this file.
+//
+//  Programmer: biagas2 -- generated by xml2avt
+//  Creation:   Wed Dec 4 14:55:21 PST 2024
+//
+// ****************************************************************************
+
+void
+avtGEOSFileReader::GetTimes(std::vector<double> &times)
+{
+    if(!haveReadRoot)
+        ReadRootFile();
+
+    times = timeSteps;
+}
+
+// ****************************************************************************
+//  Method: avtGEOSFileReader::FreeUpResources
+//
+//  Purpose:
+//      When VisIt is done focusing on a particular timestep, it asks that
+//      timestep to free up any resources (memory, file descriptors) that
+//      it has associated with it.  This method is the mechanism for doing
+//      that.
+//
+//  Programmer: biagas2 -- generated by xml2avt
+//  Creation:   Wed Dec 4 14:55:21 PST 2024
+//
+// ****************************************************************************
+
+void
+avtGEOSFileReader::FreeUpResources(void)
```

Comment:
Is there something missing here? Is this method needed now that its body is commented out?

Reply:
Thanks @markcmiller86!  Fleshed out this method.